extensions/Push/lib/Push.pm

```diff
@@ -50,15 +38,52 @@ sub start {
         $connector->backlog->reset_backoff();
     }
 
-    while(1) {
-        if ($self->_dbh_check()) {
-            $self->_reload();
-            $self->push();
+    my $pushd_loop = IO::Async::Loop->new;
+    my $main_timer = IO::Async::Timer::Periodic->new(
+        first_interval => 0,
+        interval       => POLL_INTERVAL_SECONDS,
+        reschedule     => 'drift',
+        on_tick        => sub {
+            if ( $self->_dbh_check() ) {
+                $self->_reload();
+                $self->push();
+            }
+        },
+    );
+    if ( Bugzilla->datadog ) {
+        my $dog_timer = IO::Async::Timer::Periodic->new(
+            interval   => 60 * 5,
+            reschedule => 'drift',
+            on_tick    => sub { $self->heartbeat },
+        );
+        $pushd_loop->add($dog_timer);
+        $dog_timer->start;
+    }
+
+    $pushd_loop->add($main_timer);
+    $main_timer->start;
+    $pushd_loop->run;
+}
+
+sub heartbeat {
+    my ($self) = @_;
+    my $dd = Bugzilla->datadog('pushd');
+
+    $dd->gauge('total', $self->_sql_count('SELECT COUNT(*) FROM push'));
+
+    foreach my $connector ($connectors->list) {
+        if ($connector->enabled) {
+            $dd->gauge("${name}_backlog", $self->_sql_count('SELECT COUNT(*) FROM push_backlog WHERE connector = ?', $name);
```

Comment:
Same here, "push.Phabricator_backlog". Can datadog names contain multiple dogs?

Reply:
Yes they can have dots, see the comment above.  Our convention uses snake_case.  This metric could be named `pushd.phabricator.backlog`.

In the docs we discussed measuring the age of the first queue item, so you might want to add a level to the namespace to hold both `backlog.size` and `backlog.first_item_time_enqueued`.